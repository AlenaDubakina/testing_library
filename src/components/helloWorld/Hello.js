import { useEffect, useState } from 'react';
const Hello = () => {
  const [data, setData] = useState(null);
  const [toggle, setToggle] = useState(false);
  const [value, setValue] = useState('');
  const onClick = () => {
    setToggle((prev) => !prev);
  };
  useEffect(() => {
    setTimeout(() => {
      setData({});
    }, 100);
  });
  return (
    <div className="App">
      <h1 data-testid="input-value">{value}</h1>
      {data && (
        <div style={{ color: 'red' }} data-testid="value-elem">
          data
        </div>
      )}
      {toggle && <div data-testid="toggle-elem">toggle</div>}
      <h1>Hello world!</h1>
      <button onClick={onClick} data-testid="toggle-btn">
        Click me
      </button>
      <input
        onChange={(e) => setValue(e.target.value)}
        type="text"
        placeholder="input value ..."
      />
    </div>
  );
};

export default Hello;
