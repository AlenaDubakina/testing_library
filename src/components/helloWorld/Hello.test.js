import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Hello from './Hello';

describe('Hello component', () => {
  test('renders learn react link', () => {
    render(<Hello />);
    const helloWorldElement = screen.getByText(/hello world!/i);
    const btn = screen.getByRole('button');
    const input = screen.getByPlaceholderText(/input value.../i);
    expect(helloWorldElement).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    expect(helloWorldElement).toMatchSnapshot();
  });
});

test('HelloWorld Test App', () => {
  render(<Hello />);
  const helloWorldElement = screen.queryByText(/hello2/i);
  expect(helloWorldElement).toBeNull();
});

test('DataElem App Test', async () => {
  render(<Hello />);
  const dataElem = await screen.findByTestId('value-elem');
  expect(dataElem).toBeInTheDocument();
  expect(dataElem).toHaveStyle({ color: 'red' });
});

test('Click event', () => {
  render(<Hello />);
  const btn = screen.queryByTestId('toggle-btn');
  fireEvent.click(btn);
  expect(screen.getByTestId('toggle-elem')).toBeInTheDocument();
  fireEvent.click(btn);
  expect(screen.queryByTestId('toggle-elem')).toBeNull();
  // screen.debug();
});

test('Input event', () => {
  render(<Hello />);
  const input = screen.getByPlaceholderText(/input value.../i);
  expect(screen.queryByTestId('input-value')).toContainHTML('');
  userEvent.type(input, '123123');
  expect(screen.queryByTestId('input-value')).toContainHTML('123123');
});
