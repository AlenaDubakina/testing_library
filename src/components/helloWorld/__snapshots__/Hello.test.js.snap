// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello component renders learn react link 1`] = `
<h1>
  Hello world!
</h1>
`;
